presto-hive/src/test/java/com/facebook/presto/hive/TestHiveFileFormats.java

```diff
@@ -251,6 +263,64 @@ public boolean apply(TestColumn testColumn)
     }
 
     @Test
+    public void testParquetThrift()
+            throws Exception
+    {
+        RowType nameType = new RowType(ImmutableList.of(VARCHAR, VARCHAR), Optional.empty());
+        RowType phoneType = new RowType(ImmutableList.of(VARCHAR, VARCHAR), Optional.empty());
+        RowType personType = new RowType(ImmutableList.of(nameType, BIGINT, VARCHAR, new ArrayType(phoneType)), Optional.empty());
+
+        List<TestColumn> testColumns = ImmutableList.<TestColumn>of(
+            new TestColumn(
+                "persons",
+                getStandardListObjectInspector(
+                    getStandardStructObjectInspector(
+                        ImmutableList.of("name", "id", "email", "phones"),
+                        ImmutableList.<ObjectInspector>of(
+                            getStandardStructObjectInspector(
+                              ImmutableList.of("first_name", "last_name"),
+                              ImmutableList.of(javaStringObjectInspector, javaStringObjectInspector)
+                            ),
+                            javaIntObjectInspector,
+                            javaStringObjectInspector,
+                            getStandardListObjectInspector(
+                              getStandardStructObjectInspector(
+                                ImmutableList.of("number", "type"),
+                                ImmutableList.of(javaStringObjectInspector, javaStringObjectInspector)
+                              )
+                            )
+                        )
+                    )
+                ),
+                ImmutableList.<Object>of(
+                  ImmutableList.<Object>of(
+                    ImmutableList.of("Bob", "Roberts"),
+                    0,
+                    "bob.roberts@example.com",
+                    ImmutableList.of(ImmutableList.of("1234567890"))
+                  )
+                ),
+                arraySliceOf(personType,
+                    rowSliceOf(ImmutableList.of(nameType, BIGINT, VARCHAR, new ArrayType(phoneType)),
+                        rowSliceOf(ImmutableList.of(VARCHAR, VARCHAR), "Bob", "Roberts"),
+                        0,
+                        "bob.roberts@example.com",
+                        arraySliceOf(phoneType, rowSliceOf(ImmutableList.of(VARCHAR, VARCHAR), "1234567890", null))
+                    )
+                )
+            )
+        );
+
+        InputFormat<?, ?> inputFormat = new MapredParquetInputFormat();
+        @SuppressWarnings("deprecation")
+        SerDe serde = new ParquetHiveSerDe();
+        File file = new File("src/test/resources/addressbook.parquet");
```

Comment:
Is it not possible to generate this with the record writer code that's already in this test?